docs/docs/sample-scripts/teams/export-teams-conversations.md

```diff
@@ -0,0 +1,95 @@
+- You can only retrieve a message from a Microsoft Teams team if you are a member of that team.
+
+```powershell tab="PowerShell Core"
+function  Get-Teams {
+    $teams = m365 teams team list -o json | ConvertFrom-Json -AsHashtable
+    return $teams
+}
+function  Get-Channels {
+    param (
+        [Parameter(Mandatory = $true)] [string] $teamId
+    )
+    $channels = m365 teams channel list --teamId $teamId -o json | ConvertFrom-Json -AsHashtable
+    return $channels
+}
+function  Get-Messages {
+    param (
+        [Parameter(Mandatory = $true)] [string] $teamId,
+        [Parameter(Mandatory = $true)] [string] $channelId
+    )
+    $messages = m365 teams message list --teamId $teamId --channelId $channelId -o json | ConvertFrom-Json -AsHashtable
+    return $messages
+}
+function  Get-MessageReplies {
+    param (
+        [Parameter(Mandatory = $true)] [string] $teamId,
+        [Parameter(Mandatory = $true)] [string] $channelId,
+        [Parameter(Mandatory = $true)] [string] $messageId
+    )
+
+    $messageReplies = m365 teams message reply list --teamId $teamId --channelId $channelId --messageId $messageId -o json | ConvertFrom-Json -AsHashtable
+    return $messageReplies
+}
+
+$teamsCollection = [System.Collections.ArrayList]@()
+$teams = Get-Teams
+$teamsCount = $teams.Length
+Write-Host "$teamsCount Team/s found" -ForegroundColor Magenta
+foreach ($team in $teams) {
+    $channelsCollection = [System.Collections.ArrayList]@()
+    $channels = Get-Channels $team.id
+    $teamDisplayName = $team.displayName
+    $channelsCount = $channels.Length
+    Write-Host "    $channelsCount Channel/s found in Team $teamDisplayName" -ForegroundColor Blue
+    foreach ($channel in $channels) {
+        $channelDisplayName = $channel.displayName
+        Write-Host "        Collecting conversation details from Channel $channelDisplayName" -ForegroundColor Gray
+        $messages = Get-Messages $team.id $channel.id
+        $messagesCollection = [System.Collections.ArrayList]@()
+        foreach ($message in $messages) {
+            $messageReplies = Get-MessageReplies $team.id $channel.id $message.id
+            $messageDetails = $message
+            [void]$messageDetails.Add("replies", $messageReplies)
+            [void]$messagesCollection.Add($messageDetails)
+        }
+        $channelDetails = $channel
+        [void]$channelDetails.Add("messages", $messagesCollection)
+        [void]$channelsCollection.Add($channelDetails)
+        Write-Host "        Completed" -ForegroundColor DarkGray
+    }
+    $teamDetails = $team
+    [void]$teamDetails.Add("channels", $channelsCollection)
+    [void]$teamsCollection.Add($teamDetails)
+}
+$output = @{}
+[void]$output.Add("teams", $teamsCollection)
+
+Write-Host "Creating output file" -ForegroundColor Green
+$executionDir = $PSScriptRoot
+$outputFilePath = "$executionDir/$(get-date -f yyyyMMdd-HHmmss).json"
+$output | ConvertTo-Json -Depth 10 | Out-File $outputFilePath
```

Comment:
What is the purpose of trimming JSON's depth? Could we add a comment here? My initial impression when I saw the warning in the console, was that some data was lost. If this is intentional, could we suppress the warning? A clarifying comment would also help to reinforce that this action is on purpose and no meaningful data is lost.

Reply:
@waldekmastykarz The default value of Depth is 2 so its not showing complete JSON properties. Any other ways to dynamically set this value or avoid Depth property?

Reply:
Based on the responses from the API that you have seen, what's the depth that we should use to avoid losing any data?

Reply:
@waldekmastykarz, Based on my teams conversation data, I noticed depth upto 10. Please try to execute this script against your tenant and check if you can find any keyword "System.Collections.Hashtable" in your output json file. If you find this keyword, we may need to increase the depth accordingly. Otherwise we can go ahead with 10.

Reply:
Running the script, I can see indeed a number of `Hashtable` properties so apparently we are missing some data.

I also noticed the progress bar. Nice! Would it be able to adjust it so that the top progress shows the total progress of processed Teams and then the bottom one the progress of processing channels in the current team? Right now it's still hard to tell where the script is because the progress is reset for each Team and shows only its Channels.

Reply:
@waldekmastykarz, Sure, I have included seperate progressbars. Could you please reveiw?

Reply:
Will do. What about the JSON depth?

Reply:
@waldekmastykarz, I can see depth upto 10 with all the details. Can we give a higher number and add a comment?

Reply:
@waldekmastykarz, Sure, I have updated depth value as 100 and added a note on top of that line.